editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -350,33 +331,37 @@ export const UiJsxCanvas = React.memo<UiJsxCanvasPropsWithErrorCallback>((props)
   )
   useClearSpyMetadataOnRemount(props.mountCount, props.domWalkerInvalidateCount, metadataContext)
 
-  // Handle the imports changing, this needs to run _before_ any require function
-  // calls as it's modifying the underlying DOM elements. This is somewhat working
-  // like useEffect, except that runs after everything has rendered.
-  const cssImports = useKeepReferenceEqualityIfPossible(
-    normalizedCssImportsFromImports(uiFilePath, imports),
-  )
+  const maybeOldProjectContents = React.useRef(projectContents)
+  if (ElementsToRerenderGLOBAL.current === 'rerender-all-elements') {
+    maybeOldProjectContents.current = projectContents
+  }
+
+  const maybeOldTransientFileState = React.useRef(transientFilesState)
   if (ElementsToRerenderGLOBAL.current === 'rerender-all-elements') {
-    unimportAllButTheseCSSFiles(cssImports) // TODO this needs to support more than just the storyboard file!!!!!
+    maybeOldTransientFileState.current = transientFilesState
   }
 
+  const projectContentsForRequireFn = maybeOldProjectContents.current
+  const requireFn = React.useMemo(
+    () => curriedRequireFn(projectContentsForRequireFn),
+    [curriedRequireFn, projectContentsForRequireFn],
+  )
+
+  const resolve = React.useMemo(
+    () => curriedResolveFn(projectContentsForRequireFn),
+    [curriedResolveFn, projectContentsForRequireFn],
+  )
```

Comment:
We definitely need an important note in the comment here outlining that this means the require and resolve functions will be stale if `ElementsToRerenderGLOBAL.current !== 'rerender-all-elements'`, so if we ever design a system for updating javascript blocks that also happens to run without re-rendering all elements it would require changes here (perhaps unlikely, but that would be _horrendous_ to debug)